compiler-core/src/paths.rs

```diff
@@ -45,6 +45,13 @@ pub fn build_deps_package_test(package: &str) -> PathBuf {
     build_deps_package(package).join("test")
 }
 
+/// A path to a special file that contains the version of gleam that last built
+/// the artifacts. If this file does not match the current version of gleam we
+/// will rebuild from scratch
+pub fn build_version() -> PathBuf {
+    build().join("version")
```

Comment:
Let's call it this, to avoid any potential confusion with project version
```suggestion
pub fn build_gleam_version() -> PathBuf {
    build().join("gleam_version")
```